doc/building-and-running.md

```diff
@@ -0,0 +1,76 @@
+MAGDA can be run in two main ways:
+
+* Kubernetes - Each component is built into a Docker container and run on a Kubernetes cluster, which may be a robust production environment or a simple local cluster like Minikube.  This is good for production environments and for testing in a more production-like environment on your development machine.  It also creates a more consistent environment, potentially avoiding configuration quirks of your development machine.
+* Host - The components are run in JVM processes running directly on your development machine.  This allows for fast development cycles because a code change can quickly trigger a restart of the JVM running that component. 
+
+## Kubernetes (local, minikube)
+
+Running on a local Kubernetes cluster involves these steps:
+
+1. Set up your cluster.  On a development system, this means installing minikube and kubectl.
+2. Deploy the kube registry to the cluster.
+
+```bash
+kubectl create -f deploy/kubernetes/registry.yml
+```
+
+3. Compile the components, create their Docker images, and push them to the local registry.  This will also terminate the running pods associated with the components, triggering them to restart with the new image.
+
+```bash
+sbt deployLocal
+```
+
+4. Deploy to the cluster
+ 
+```bash
+cd deploy/kubernetes
+kubectl create -f elasticsearch.yml -f registry-datastore.yml -f local.yml
```

Comment:
Need to tell them to build the es docker image unless we can get rid of it for 5.0